Document to ignore drift detection (#4288)
pipe-cd/pipecd: docs/content/en/docs-dev/user-guide/managing-application/configuration-drift-detection.md

This feature is automatically enabled for all applications.

You can change the checking interval as well as [configure the notification](../../managing-piped/configuring-notifications/) for these events in `piped` configuration.

### Ignore drift detection for specific fields

>  Note: This feature is currently supported for only Kubernetes application.  

You can also ignore drift detection for specified fields in your application manifests. In other words, even if the selected fields have different values between live state and Git, the application status will not be set to `Out of Sync`.

> To use this feature, set parameters in the application configuration. For more information, see the [configuration reference](../../configuration-reference/#driftdetection) and the following `Example` section.

Reviewer: ```suggestion
```

> #### Example

Reviewer: ```suggestion
For example, suppose you have the application's manifest as below 
```

apiVersion: apps/v1
kind: Deployment
metadata:
  name: simple
spec:
  replicas: 2
  template:
    spec:
      containers:
        - args:
            - hi
            - hello
          image: gcr.io/pipecd/helloworld:v1.0.0
          name: helloworld
```

If you want to ignore the drift detection for the two sceans
- pod's replicas
- `helloworld` container's args

> Add the following statements to `app.pipe.yaml` to ignore diff on those fields.

Reviewer: ```suggestion
Add the following statements to `app.pipecd.yaml` to ignore diff on those fields.
```

> you may add the follow statements to `app.pipe.yaml`.

Reviewer: ```suggestion
Add the following statements to `app.pipe.yaml` to ignore diff on those fields.
```

```yaml
spec:
  ...
  driftDetection:
    ignoreFields:
      - spec.replicas
      - spec.template.spec.containers.0.args
```

> 

Reviewer: ```suggestion
For more information, see the [configuration reference](../../configuration-reference/#driftdetection).
```

This feature is automatically enabled for all applications.

You can change the checking interval as well as [configure the notification](../../managing-piped/configuring-notifications/) for these events in `piped` configuration.

> ### Enable to ignore drift detection for specific fields

Reviewer: ```suggestion
### Ignore drift detection for specific fields
```

> >  Note: This feature is currently supported for only kubernetes application.  

Reviewer: ```suggestion
>  Note: This feature is currently supported for only Kubernetes application.  
```

> This feature ignores drift detection for the specified fields. In other words, even if the specified fields are different between live state and Git, the application status will not be `Out of Sync`.

Reviewer: ```suggestion
You can also ignore drift detection for specified fields in your application manifests. In other words, even if the selected fields have different values between live state and Git, the application status will not be set to `Out of Sync`.
```

To use this feature, set parameters in the application configuration. For more information, see the [configuration reference](../../configuration-reference/#driftdetection) and the following `Syntax` section.

#### Syntax
There are two kinds of block collection in yaml.

Sequence: This block collection is indicated each entry with a dash and space ("`- `").  
Mapping: This block collection is indicated each entry with a colon and space("`: `").

Mapping indexes are specified, for example `.foo`.  
Sequence indexes are specified, for example `.2`.  
However, the first index doesn't need the `.`.

> 

Reviewer: I think we don't need this blog, users are expected to understand `yaml` syntax or if not we have to add this to every other places 👀

Author: I see. I will remove this section. 
Thank you:smile:

Reviewer: Also, should move L63 under the example as well

Author: I fixed it 😃
